### PR TITLE
fix: show exc info by default

### PR DESCRIPTION
**Pull request by JoanFM**
**Changes introduced**
Productivity boost to capture faster exceptions and problems when defining problems. Also easier for the community to see the problems and to have better reporting.

@cristianmtr

### CODE REVIEW

**Review by theUnkownName — DISMISSED** [review]

File: jina/peapods/peas/__init__.py [hunk 1]
```diff
@@ -78,18 +78,18 @@ def run(self):
             except KeyboardInterrupt:
                 self.logger.info(f'{self.runtime!r} is interrupted by user')
             except (Exception, SystemError) as ex:
-                self.logger.error(f'{ex!r} during {self.runtime.run_forever!r}' +
-                                  f'add "--show-exc-info" to see the exception stack in details'
-                                  if not self.args.show_exc_info else '',
-                                  exc_info=self.args.show_exc_info)
+                self.logger.error(f'{ex!r} during {self.runtime_cls.__init__!r}' +
```
Comment (hanxiao) [inline jina/peapods/peas/__init__.py:81]:
```suggestion
                self.logger.error(f'{ex!r} during {self.runtime_cls.__init__!r}' +
```
copy paste glitch

File: jina/peapods/peas/__init__.py [hunk 2]
```diff
@@ -65,10 +65,10 @@ def run(self):
         try:
             self.runtime.setup()
         except Exception as ex:
-            self.logger.error(f'{ex!r} during {self.runtime.setup!r}' +
-                              f'add "--show-exc-info" to see the exception stack in details'
-                              if not self.args.show_exc_info else '',
-                              exc_info=self.args.show_exc_info)
+            self.logger.error(f'{ex!r} during {self.runtime_cls.__init__!r}' +
```
Comment (hanxiao) [inline jina/peapods/peas/__init__.py:68]:
```suggestion
            self.logger.error(f'{ex!r} during {self.runtime_cls.__init__!r}' +
```
copy paste glitch

File: jina/peapods/peas/__init__.py [hunk 3]
```diff
@@ -78,18 +78,18 @@ def run(self):
...
+                                  exc_info=not self.args.hide_exc_info)
 
             try:
                 self.runtime.teardown()
             except Exception as ex:
-                self.logger.error(f'{ex!r} during {self.runtime.teardown!r}' +
-                                  f'add "--show-exc-info" to see the exception stack in details'
-                                  if not self.args.show_exc_info else '',
-                                  exc_info=self.args.show_exc_info)
+                self.logger.error(f'{ex!r} during {self.runtime_cls.__init__!r}' +
```
Comment (hanxiao) [inline jina/peapods/peas/__init__.py:89]:
```suggestion
                self.logger.error(f'{ex!r} during {self.runtime_cls.__init__!r}' +
```
copy paste glitch

File: jina/peapods/peas/__init__.py [hunk 4]
```diff
@@ -136,10 +137,10 @@ def close(self) -> None:
                 self.runtime.cancel()
                 self.is_shutdown.wait()
             except Exception as ex:
-                self.logger.error(f'{ex!r} during {self.runtime.cancel!r}' +
-                                  f'add "--show-exc-info" to see the exception stack in details'
-                                  if not self.args.show_exc_info else '',
-                                  exc_info=self.args.show_exc_info)
+                self.logger.error(f'{ex!r} during {self.runtime_cls.__init__!r}' +
```
Comment (hanxiao) [inline jina/peapods/peas/__init__.py:140]:
```suggestion
                self.logger.error(f'{ex!r} during {self.runtime_cls.__init__!r}' +
```
cpg

File: jina/parsers/peapods/base.py [hunk 5]
```diff
@@ -34,6 +34,6 @@ def mixin_base_ppr_parser(parser):
                     help='A UUID string to represent the identity of this object'
                     if _SHOW_ALL_ARGS else argparse.SUPPRESS)
 
-    gp.add_argument('--show-exc-info', action='store_true', default=False,
+    gp.add_argument('--hide-exc-info', action='store_true', default=True,
```
Comment (hanxiao) [inline jina/parsers/peapods/base.py:37]:
```suggestion
    gp.add_argument('--hide-exc-info', action='store_true', default=True,
```
default=True???

**Review by hanxiao — CHANGES_REQUESTED** [review]
see comments

File: jina/parsers/peapods/base.py (hunk 5 above)
Comment (JoanFM) [reply jina/parsers/peapods/base.py:37]: oh my bad

Comment (JoanFM) [reply jina/parsers/peapods/base.py:37]:
```suggestion
    gp.add_argument('--hide-exc-info', action='store_true', default=False,
```

File: jina/peapods/peas/__init__.py (hunk 2 above)
Comment (JoanFM) [reply jina/peapods/peas/__init__.py:68]:
```suggestion
            self.logger.error(f'{ex!r} during {self.runtime.setup!r}' +
```

File: jina/peapods/peas/__init__.py (hunk 3 above)
Comment (JoanFM) [reply jina/peapods/peas/__init__.py:89]:
```suggestion
                self.logger.error(f'{ex!r} during {self.runtime.teardown!r}' +
```

File: jina/peapods/peas/__init__.py (hunk 4 above)
Comment (JoanFM) [reply jina/peapods/peas/__init__.py:140]:
```suggestion
                self.logger.error(f'{ex!r} during {self.runtime.cancel!r}' +
```

File: jina/peapods/peas/__init__.py (hunk 1 above)
Comment (JoanFM) [reply jina/peapods/peas/__init__.py:81]:
```suggestion
                self.logger.error(f'{ex!r} during {self.runtime.run_forever!r}' +
```

File: jina/peapods/peas/__init__.py [hunk 6]
```diff
@@ -111,8 +111,9 @@ def start(self):
         if self.ready_or_shutdown.wait(_timeout):
             if self.is_shutdown.is_set():
                 # return too early and the shutdown is set, means something fails!!
-                self.logger.critical(f'fail to start {self!r} because {self.runtime!r} throws some exception, '
-                                     f'add "--show-exc-info" to see the exception stack in details')
+                if self.args.hide_exc_info:
+                    self.logger.critical(f'fail to start {self!r} because {self.runtime!r} throws some exception, '
+                                         f'remote "--hide-exc-info" to see the exception stack in details')
```
Comment (JoanFM) [inline jina/peapods/peas/__init__.py:116]:
```suggestion
                                         f'remove "--hide-exc-info" to see the exception stack in details')
```

Comment (JoanFM) [inline jina/peapods/peas/__init__.py:116]:
```suggestion
                                         f'remove "--hide-exc-info" to see the exception stack in details')
```

**Review by hanxiao — APPROVED** [review]

**Pull request merged**